uswds/uswds-compile
# USWDS-Compile - POAM: December '24

# Summary

POAM dependency updates for November + December 2024.

## Related issue

[USWDS-Team - POAM: December 2024](https://github.com/uswds/uswds-team/issues/413)

## Vulnerabilities

No changes in vulnerabilities

```node
found 0 vulnerabilities
```

## Major changes

- Updated to latest USWDS version
- New `sass-emedded` version brings on new sass deprecation warnings

### New sass deprecation warnings

We have a handful of new sass warnings that are repeated throughout build process:

```jsx
Deprecation Warning: Sass @import rules are deprecated and will be removed in Dart Sass 3.0.0.

Deprecation Warning: Global built-in functions are deprecated and will be removed in Dart Sass 3.0.0.
Use math.unit instead.

Deprecation Warning: Global built-in functions are deprecated and will be removed in Dart Sass 3.0.0.
Use map.merge instead.

Deprecation Warning: Global built-in functions are deprecated and will be removed in Dart Sass 3.0.0.
Use string.unquote instead.

Deprecation Warning: color.lightness() is deprecated. Suggestion: color.channel($color, "lightness", $space: hsl)
```

Deprecations were added to uswds/uswds#6104 to resolve. We can also chose to silence these deprecations for downstream users if we would like, or alternatively, we can choose to not update sass until deprecations are handled.

## Dependency updates

| Dependency Name | Old Version | New Version |
| --- | --- | --- |
| @uswds/uswds | ^3.9.0 | ^3.10.0 |
| postcss | 8.4.47 | 8.4.49 |
| sass-embedded | 1.79.4 | 1.83.0 |

## Testing instructions

> [!TIP]
> To reduce review time, I have updated the November PR which had not been merged yet. Because of this, you'll see `cm-poam-nov-24` as the branch name. I have updated the branch with the latest changes from December.

1. Install this branch on site
    - On Site's `main` branch, run: 
      ```bash
      npm install "https://github.com/uswds/uswds-compile/tree/cm-poam-nov-24" --save
      ```
2. Run gulp build scripts and ensure compile runs as expected without error.
    - You'll see new deprecation warnings. These deprecated functions will be supported until the next major version of Sass is supported.
3. Ensure no regressions when package is used to compile site
